burst/src/main/java/com/squareup/burst/Burst.java

```
@@ -46,20 +49,51 @@
    * not an enum value.
    */
   public static String friendlyName(Enum<?>[] arguments) {
+    return friendlyName(arguments, null);
+  }
+
+  /**
+   * Creates an "exploded" test name which includes information about the {@code constructorArgs}
+   * and {@code methodArgs}. This will append both the enum class and enum value name for every
+   * argument in order.
+   * <p>
+   * For example, a method named "snackBreak" being invoked with constructor arguments
+   * {@code Drink.SODA} and {@code Snack.ALMONDS} and method arguments {@code BreakTime.AFTERNOON}
+   * would produce "snackBreak_DrinkSODA_SnackALMONDS_BreakTimeAFTERNOON".
+   * <p>
+   * If any of the arguments have an {@literal @}Name annotation, the enum class name will be
+   * replaced with the value provided in the annotation.
+   *
+   * @throws ClassCastException If any element of {@code constructorArgs} or {@code methodArgs} is
+   * not an enum value.
+   */
+  public static String friendlyName(Enum<?>[] arguments, Annotation[][] argumentAnnotations) {
     checkNotNull(arguments, "arguments");
     if (arguments.length == 0) {
       return "";
     }
 
     StringBuilder builder = new StringBuilder();
-    for (Object argument : arguments) {
+    for (int i = 0; i < arguments.length; i++) {
+      Object argument = arguments[i];
       if (builder.length() > 0) {
         builder.append(", ");
       }
 
       Enum<?> value = (Enum<?>) argument;
       // Appends the enum name and value name. (e.g., Card.VISA)
-      builder.append(value.getClass().getSimpleName()).append('.').append(value);
+      if (argumentAnnotations != null
+          && argumentAnnotations.length > i
```

`arguments.length` should always equal `argumentAnnotations.length` unless we have a bug right? If there was a bug which made that not the case, I'd rather we fail hard.